package/MDAnalysis/core/Selection.py

```diff
@@ -178,6 +178,12 @@ def __init__(self, parser, tokens):
         pass
 
     def apply(self, group):
+        # Check whether group is identical to the one stored
+        # in the corresponding universe, in which case this
+        # is returned directly. This works since the Universe.atoms
+        # are unique by construction.
+        if group.universe and group is group.universe.atoms:
```

Comment:
Actually, it is doing what I think it is :), although I don't think it is necessary. I test for whether `group.universe` is different from `None` before reading `group.universe.atoms` just to be sure I don't get an `AttributeError`. As far as I could see, group.universe is always set, however, I became uncertain after seeing this definition:
```python
    @property
    def dimensions(self):
        """Dimensions of the Universe to which the group belongs, at the current time step."""
        if self.universe is not None:
            return self.universe.dimensions
        else:
            raise AttributeError("This AtomGroup does not belong to a Universe with a dimension.")
```
Therefore, I added this extra check (although I agree it should perhaps have been made more explicit by writing `group.universe is not None`. I would be happy to change it to what you suggest, if universe is indeed always set.